[CIAPP-7313] Create documentation for GitHub checks in Quality Gates (#19474)
DataDog/documentation: content/en/continuous_integration/quality_gates/_index.md

{{< img src="ci/delete_quality_gate_rule.png" alt="Delete a Quality Gates rule" style="width:90%;">}}

> ## Enable GitHub checks creation

Author: I had this one as plural as generally customers have more than 1 rule so several checks will be created. Do you think it would still be better to have it singular?

> ## Enable GitHub checks creation

Reviewer: Yes, I think the singular form is better here. If you wanted to use a plural, a better construction would be "Enable creation of GitHub checks", but the singular form here is fairly common and I think the user will understand what we mean

> ## Enable GitHub checks creation

Author: Perfect, thanks! I've committed the change then

> A [GitHub check][9] can be automatically created for each rule evaluated. When this feature is enabled, you will see the evaluation results directly in GitHub.
> The check contains additional information about the rule evaluation, such as the failure reason and the matching events in Datadog.

Reviewer: ```suggestion
You can automatically create a [GitHub check][9] for each rule evaluated. When this feature is enabled, the evaluation results appear directly in GitHub.
The check contains additional information about the rule evaluation, such as the failure reason and the matching events in Datadog.
```

{{< img src="ci/delete_quality_gate_rule.png" alt="Delete a Quality Gates rule" style="width:90%;">}}

## Enable GitHub checks creation

[GitHub checks][9] can be automatically created on the commits in which the quality gates rules are evaluated. This allows you to see the results of

> the rule evaluations directly in GitHub. When this feature is enabled, a GitHub check will be created for each rule evaluated.

Author: Looks good to me 👍 Thanks for the feedback!

The check contains additional information about the rule evaluation, such as the failure reason and the matching events in Datadog.

> To enable this feature:

Reviewer 2: ```suggestion
To enable GitHub Checks:
```

[6]: https://app.datadoghq.com/organization-settings/application-keys
[7]: https://github.com/DataDog/datadog-ci/blob/master/README.md
[8]: /continuous_integration/guides/flaky_test_management/

> [9]: https://docs.github.com/en/rest/checks?apiVersion=2022-11-28

Author: I checked and we can, I'll commit the change
